### PR TITLE
[SR GeekCamp] Support function array_cum_sum in StarRocks

### LINKED ISSUE #4016: Support function array_cum_sum in StarRocks【Easy】
# Requirement Description
SELECT array_cum_sum([1, 1, 1, 1]) AS res

```
 [1, 2, 3, 4] 
```

# Mentor
@stdpain 

# Registration procedure

**Step 1.** Add a comment when you sign up. The comment can be a brief self-introduction and reason for signing up, including what you do, what you're good at, what you're interested in and your technology stack, and how familiar you are with StarRocks. Example:

> I pick! My name is Johnny. I am from StarRocks and I have been working on database kernel development for 1 year. I am interested in OLAP, good at C++ programming, and have been using StarRocks for 6 months. I chose to solve this issue because I am familiar with execution engine.

**Step 2.** On your WeChat, add the StarRocks community assistant "**_StarRocks-1_**" as a WeChat friend and leave your GitHub ID. The assistant will invite you to a WeChat group based on the issue you picked.

**Step 3.** The mentor will communicate with all the participants who picked this issue and assign the task to the best match.

# Before you start
1. Read the StarRocks [GitHub workflow](https://github.com/StarRocks/community/blob/main/Contributors/guide/workflow.md) carefully, which provides the configurations of BE and FE and describes how to submit a pull request.
2. To learn more about StarRocks, read [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).

# Pull request requirements

1. After you complete code development, upload the code to the StarRocks code repo as a pull request (PR).

  - PR title format: **[SR GeekCamp] + PR title**
  - Labels ：[Geek Camp 2022/feature]
2. After you submit your PR, comment on the status of your PR submission as requested under the Pinned Issue [[StarRocks geek Camp 2022] ](https://github.com/StarRocks/starrocks/issues/4008)page.

  - Comment format: **"I have finished Issue #" + PR link id**

### PR DESCRIPTION
**Pull request by sym-liuyang**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：
fix #4016

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: gensrc/script/vectorized/vectorized_functions.py [hunk 1]
```diff
@@ -647,6 +647,7 @@
     [150139, 'reverse', 'ARRAY_DECIMALV2', ['ARRAY_DECIMALV2'], 'ArrayFunctions::array_reverse_decimalv2'],
     [150140, 'reverse', 'ARRAY_DATETIME',  ['ARRAY_DATETIME'],  'ArrayFunctions::array_reverse_datetime'],
     [150141, 'reverse', 'ARRAY_DATE',      ['ARRAY_DATE'],      'ArrayFunctions::array_reverse_date'],
+    [150142, 'array_cum_sum', 'ANY_ARRAY', ['ANY_ARRAY'], 'ArrayFunctions::array_cum_sum'],
```
Comment (kangkaisen) [inline gensrc/script/vectorized/vectorized_functions.py:650]: I think support ARRAY_BIGINT and ARRAY_DOUBLE is enough.

**Comment by kangkaisen** [comment]
need add BE UT.

File: be/src/exprs/vectorized/array_functions.cpp [hunk 2]
```diff
@@ -338,6 +338,130 @@ class ArrayRemoveImpl {
...
+    const ColumnPtr& arg0 = columns[0];
+    // TODO: dispatch graceful
+
+    ConstColumn* input = down_cast<ConstColumn*>(arg0.get());
+    auto arr_col_h = input->data_column()->clone();
+    ArrayColumn* arr_col = down_cast<ArrayColumn*>(arr_col_h.get());
+    auto* element = arr_col->elements_column().get();
+    element = element->is_nullable() ? down_cast<NullableColumn*>(element)->data_column().get() : element;
+
+#define HANDLE_ELEMENT_TYPE(ElementType, TYPE)            \
```
Comment (kangkaisen) [inline be/src/exprs/vectorized/array_functions.cpp:447]: repeated define macro. please rename.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by stdpain — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by stdpain** [comment]
UT will be added later. You could add a document for this function in https://github.com/StarRocks/docs.zh-cn/tree/main/sql-reference/sql-functions/array-functions @sym-liuyang

**Pull request merged**